src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt

```diff
@@ -104,8 +114,23 @@ fun <T> emptyWhenNotFound(exception: WebClientResponseException): Mono<T> = empt
 fun <T> emptyWhen(exception: WebClientResponseException, statusCode: HttpStatus): Mono<T> =
   if (exception.statusCode == statusCode) Mono.empty() else Mono.error(exception)
 
-fun <T> propagateClientError(exception: WebClientResponseException, message: String): Mono<T> =
-  if (exception.statusCode.is4xxClientError) Mono.error(ClientException(exception, message)) else Mono.error(exception)
+fun <T> propagateClientError(
+  exception: WebClientResponseException,
+  message: String
+): Mono<T> = (
+  if (exception.statusCode.is4xxClientError) {
+    val mapper = jacksonObjectMapper()
```

Comment:
These jackson instances are pretty heavyweight but threadsafe so generally advised to have one as a field that is reused. 
This way of creating the mapper is inconsistent with how the mapper is configured by spring boot - should be able to just inject the mapper bean in via the constructor.

Alternatively the other methods on the flux like [onStatus](https://www.baeldung.com/spring-webclient-get-response-body#using-onstatus) or [filter](https://www.baeldung.com/spring-webclient-get-response-body#using-exchangefilterfunction) provide a client response which provides a body to mono method which uses the same jackson mapper underneath and mean you don't need to reference it at all: 

```
.onStatus(
    HttpStatus.INTERNAL_SERVER_ERROR::equals,
    response -> response.bodyToMono(String.class)
```